modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1547 @@
+    }
+
+    @Test(dataProvider = "dataForTestCreateBadRequestException")
+    public void testCreateBadRequestException(String scimObjectString, String attributes,
+                                              String excludeAttributes,
+                                              int expectedScimResponseStatus)
+            throws BadRequestException, NotFoundException, CharonException, ConflictException {
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+        when(userManager.createMe(anyObject(), anyObject())).thenThrow(BadRequestException.class);
+
+        SCIMResponse outputScimResponse = meResourceManager.create(scimObjectString,
+                userManager, attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForTestCreateConflictException")
+    public Object[][] dataToTestCreatConflictException() {
+
+        return new Object[][]{
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", null, ResponseCodeConstants.CODE_CONFLICT}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateConflictException")
+    public void testCreateConflictException(String scimObjectString, String attributes,
+                                            String excludeAttributes,
+                                            int expectedScimResponseStatus)
+            throws BadRequestException, NotFoundException, CharonException, ConflictException {
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(ConflictException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new ConflictException()));
+        when(userManager.createMe(anyObject(), anyObject())).thenThrow(ConflictException.class);
+
+        SCIMResponse outputScimResponse = meResourceManager.create(scimObjectString, userManager,
+                attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForTestCreateNotFoundException")
+    public Object[][] dataToTestCreateNotFoundException() {
```

Comment:
```suggestion
    public Object[][] dataToTestCreateUserNotFoundException() {
```

change the data provider name appropriately

Reply:
Fixed in commit a5699e694bc31369fb4a347d4a0a12177e73775d